sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala

```diff
@@ -225,59 +191,71 @@ object AlluxioUtils extends Logging {
   // Suppose the key doesn't exist when using like Databricks's instance profile
-  private def autoMountBucket(
-      scheme: String,
-      bucket: String,
-      access_key: Option[String],
-      secret_key: Option[String]): Unit = {
+  private def autoMountBucket(alluxioUser: String, scheme: String, bucket: String,
+      s3AccessKey: Option[String], s3SecretKey: Option[String]): Unit = {
+    val conf = getS3ClientConf(alluxioUser, s3AccessKey, s3SecretKey)
+
     // to match the output of alluxio fs mount, append / to remote_path
     // and add / before bucket name for absolute path in Alluxio
     val remote_path = scheme + bucket + "/"
     val local_bucket = "/" + bucket
     this.synchronized {
       if (!mountedBuckets.contains(local_bucket)) {
-        // not mount yet, call mount command
-        // we only support s3 or s3a bucket for now.
-        // To support other cloud storage, we need to support credential parameters for the others
-        val parameter = "fs mount --readonly " +(
-          if (access_key.isDefined && secret_key.isDefined) {
-            s"--option s3a.accessKeyId=${access_key.get} " +
-            s"--option s3a.secretKey=${secret_key.get} " } else "") +
-          s"$local_bucket $remote_path"
-
-        val (output, _) = runAlluxioCmd(parameter)
-        if (output != 0) {
-          throw new RuntimeException(s"Mount bucket $remote_path to $local_bucket failed $output")
+        try {
+          withResource(alluxio.client.file.FileSystem.Factory.create(conf)) { fs =>
+            // not mount yet, call mount command
+            // we only support s3 or s3a bucket for now.
+            // To support other cloud storage,
+            // we need to support credential parameters for the others
+            val mountOptionsBuilder = MountPOptions.newBuilder().setReadOnly(true)
+            s3AccessKey.foreach(e => mountOptionsBuilder.putProperties("s3a.accessKeyId", e))
+            s3SecretKey.foreach(e => mountOptionsBuilder.putProperties("s3a.secretKey", e))
+            fs.mount(new AlluxioURI(local_bucket), new AlluxioURI(remote_path),
+              mountOptionsBuilder.build())
+            logInfo(s"Mounted bucket $remote_path to $local_bucket in Alluxio")
+            mountedBuckets(local_bucket) = remote_path
+          }
+        } catch {
+          case NonFatal(e) =>
+            throw new RuntimeException(s"Mount bucket $remote_path to $local_bucket failed", e)
         }
-        logInfo(s"Mounted bucket $remote_path to $local_bucket in Alluxio $output")
-        mountedBuckets(local_bucket) = remote_path
       } else if (mountedBuckets(local_bucket).equals(remote_path)) {
         logDebug(s"Already mounted bucket $remote_path to $local_bucket in Alluxio")
       } else {
         throw new RuntimeException(s"Found a same bucket name in $remote_path " +
-          s"and ${mountedBuckets(local_bucket)}")
+            s"and ${mountedBuckets(local_bucket)}")
```

Comment:
I would expect spacing to be 2 spaces here like it was before

Reply:
Done.